File: jsimple-json/c#/jsimple/json/objectmodel/JsonNull.cs
namespace jsimple.json.objectmodel
{

	/// <summary>
	/// This class is used to represent a "null" literal in JSON.  By having an explicit class, instead of just the regular
	/// Java null, we can differentiate more easily between an object not existing in the JSON and it existing and having the
	/// explicit JSON literal value "null".
	/// 
	/// @author Bret Johnson
	/// @since 5/20/12 9:52 PM
	/// </summary>
	// TODO: Remove this & just treat as null
	public sealed class JsonNull
	{
		public static JsonNull singleton = new JsonNull();

		private JsonNull()
		{
		}

		public override string ToString()
		{
			return "JSON null primitive";
		}
	}

}

File: jsimple-json/src/main/java/jsimple/json/objectmodel/JsonNull.java
/*
 * Copyright (c) 2012-2014, Microsoft Mobile
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */

package jsimple.json.objectmodel;

/**
 * This class is used to represent a "null" literal in JSON.  By having an explicit class, instead of just the regular
 * Java null, we can differentiate more easily between an object not existing in the JSON and it existing and having the
 * explicit JSON literal value "null".
 *
 * @author Bret Johnson
 * @since 5/20/12 9:52 PM
 */
// TODO: Remove this & just treat as null
public final class JsonNull {
    static public JsonNull singleton = new JsonNull();

    private JsonNull() {
    }

    @Override public String toString() {
        return "JSON null primitive";
    }
}
